paddle/fluid/ir_adaptor/translator/op_translator.cc

```diff
@@ -1733,96 +1582,61 @@ struct MulGradOpTranscriber : public OpTranscriber {
                  "Expected op[%s]'s input %s is DenseTensor but got %s",
                  op_desc.Type(),
-                 x_grad_var_name,
-                 x_grad_type);
-      dialect::DenseTensorType x_grad_tensor_type =
-          x_grad_type.dyn_cast<dialect::DenseTensorType>();
+                 grad_var_name,
+                 grad_type);
+      dialect::DenseTensorType grad_tensor_type =
+          grad_type.dyn_cast<dialect::DenseTensorType>();
+
+      VLOG(10) << "[" << op_desc.Type() << "] shape of " << var_name
+               << " change from " << grad_tensor_type.dims() << " to " << dim;
 
-      VLOG(10) << "[" << op_desc.Type() << "] x_grad_shape change from "
-               << x_grad_tensor_type.dims() << " to " << dim_x;
+      param_map->PushValue(grad_var_name,
+                           VariableDefiningInfo(reshape_op.out(), false, -1));
+    };
 
-      param_map->PushValue(x_grad_var_name,
-                           VariableDefiningInfo(reshape_op_x.out(), false, -1));
+    if (x_grad_output.size()) {
+      gradReshape("X@GRAD");
     }
 
     if (y_grad_output.size() < 1) {
       return;
     }
-    IR_ENFORCE(
-        y_grad_output.size() == 1,
-        "Expected op[%s]'s output Y@GRAD has only 1 variable, but got %d",
-        op_desc.Type(),
-        y_grad_output.size());
-    const auto& y_grad_var_name = y_grad_output[0];
 
-    auto idx_iter_y = arg_to_idx.find(y_grad_var_name);
-    if (idx_iter_y == arg_to_idx.end()) {
-      IR_THROW("op[%s] should have got its y_grad", op_desc.Type());
-    }
-    auto [idx_in_op_y, idx_in_vec_y] = idx_iter_y->second;
-    VLOG(10) << "[output recording]"
-             << "[" << op_desc.Type() << "]" << y_grad_var_name << " "
-             << idx_in_op_y << " " << idx_in_vec_y;
-
-    VarDesc* var_desc_y = op_desc.Block()->FindVarRecursive("Y");
-    std::vector<int64_t> y_shape = var_desc_y->GetShape();
-    DenseTensorTypeStorage::Dim dim_y = phi::make_ddim(y_shape);
-
-    pir::OpResult y_value_res = operation->result(idx_in_op_y);
-
-    auto reshape_op_y = builder.Build<dialect::ReshapeOp>(y_value_res, y_shape);
-
-    IR_ENFORCE(y_value_res,
-               "Expected op[%s]'s input %s is not null",
-               op_desc.Type(),
-               y_grad_var_name);
-    pir::Type y_grad_type = y_value_res.type();
-    IR_ENFORCE(y_grad_type.isa<dialect::DenseTensorType>(),
-               "Expected op[%s]'s input %s is DenseTensor but got %s",
-               op_desc.Type(),
-               y_grad_var_name,
-               y_grad_type);
-    dialect::DenseTensorType y_grad_tensor_type =
-        y_grad_type.dyn_cast<dialect::DenseTensorType>();
-
-    VLOG(10) << "[" << op_desc.Type() << "] y_grad_shape change from "
-             << y_grad_tensor_type.dims() << " to " << dim_y;
-
-    param_map->PushValue(y_grad_var_name,
-                         VariableDefiningInfo(reshape_op_y.out(), false, -1));
+    gradReshape("Y@GRAD");
```

Comment:
```suggestion
   if (y_grad_output.size()) {
      gradReshape("Y@GRAD");
   }
```
Keeping consistent form may be better.